doc_src/cmds/string-match.rst

```diff
@@ -27,7 +27,7 @@ If ``--index`` or ``-n`` is given, each match is reported as a 1-based start pos
 
 If ``--regex`` or ``-r`` is given, PATTERN is interpreted as a Perl-compatible regular expression, which does not have to match the entire STRING. For a regular expression containing capturing groups, multiple items will be reported for each match, one for the entire match and one for each capturing group. With this, only the matching part of the STRING will be reported, unless ``--entire`` is given.
 
-When matching via regular expressions, it is possible to directly import matches as fish variables by means of named capture groups using the PCRE2 syntax. This behavior is automatic and occurs in addition to any other match reporting. The default behavior with ``--regex`` results in the initialization of fish variables in the default scope containing the matched text corresponding to each named capture group. A named capture group matching a zero-length string will be initialized as a fish variable containing a likewise empty string (i.e. the equivalent of ``""``), but a named capture group that did not match will result in an empty (null) fish variable. When ``--regex`` is used in conjunction with ``--all``, this behavior changes slightly: for each of the *n* matching sequences, the *n*th index of each named variable will contain the contents of the corresponding named capture group for the *n*th match, but will contain an empty string if the group was empty or if the group did not match.
```

Comment:
The literal space looks a little funky: `*n* th`

<img width="821" alt="nth-spaces" src="https://user-images.githubusercontent.com/5209296/100703556-61c7d180-3369-11eb-91b8-111baf69c663.png">

Escaping looks slightly better to me: `*n*\ th`

<img width="827" alt="nth-escapes" src="https://user-images.githubusercontent.com/5209296/100703555-612f3b00-3369-11eb-8472-55d0779db533.png">

A superscript looks cool but messes up the line height:
```
*n*:superscript:`th`
```

<img width="825" alt="nth-superscripts" src="https://user-images.githubusercontent.com/5209296/100703554-5ffe0e00-3369-11eb-9d3d-2b9d1f510f69.png">